src/React/Widgets/AnnotationStoreEditorWidget/index.js

```diff
@@ -41,7 +42,8 @@ export default function annotationStoreEditorWidget(props) {
 
   if (props.annotation && props.annotations[props.annotation.id]) {
     const storedSelectedAnnotation = props.annotations[props.annotation.id];
-    if (storedSelectedAnnotation.generation === props.annotation.generation) {
+    if (storedSelectedAnnotation.generation === props.annotation.generation ||
+        deepEquals(Object.assign({}, storedSelectedAnnotation, { generation: props.annotation.generation }), props.annotation)) {
```

Comment:
So would I do it like this?
`    if ((!this.props.ignoreGeneration && storedSelectedAnnotation.generation === props.annotation.generation) ||
        (this.props.ignoreGeneration &&
         deepEquals(Object.assign({}, storedSelectedAnnotation, { generation: props.annotation.generation }), props.annotation))) {
`

I was thinking of the generation as a shortcut - when it's equal, you know you don't have to do the more expensive comparison. I wasn't thinking of it as a way to ignore other changes in the annotation - so what I implemented above/your proposal is a bit different. I'm not sure which might be better?

To be complete - my idea with a prop would be:
`if (storedSelectedAnnotation.generation === props.annotation.generation ||
        (this.props.ignoreGeneration && deepEquals(...))`